### ISSUE TITLE
Lags and app crash if have two 1-1 chats [release]

### ISSUE THREAD

**Issue by annadanchenko** [status-im/status-react#2990]
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: There are 2 1-1 chats and each has 2 messages. After app relaunch (kill it and run again) and doing nothing, just waiting for about 3 minutes on my real device Galaxy S6 it crashes. If try to open chats and go back to Chats and switch between tabs - it's slow and then eventually crashes. 

There is no crash reported in TF as it looks like system just shutdowns app and it goes to background and screen for Sign in is shown if switch to it. From TF session I see that memory consumption increases constantly

Happens on build with JSC and without (release build) and on Jan 10 nightly build.

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
no crash, can switch to chats, tabs
#### Actual behavior
[comment]: # (Describe what actually happened.)
crash
### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)
Android app, Release build (no JSC) https://app.testfairy.com/projects/4803622-status/builds/7469768/sessions/9/?accessToken=gy3s4De0SG3UBizpVoURav8uLOs
Android app, Nightly build: https://app.testfairy.com/projects/4803622-status/builds/7463949/sessions/14/?accessToken=VVZbVNbNBTmnjTgmeUiya5EanvM

iOS app, release build no log only memory graph: https://app.testfairy.com/projects/4803590-status/builds/7470662/sessions/2/?accessToken=sQQ70YkbH8thkArO8FytK--IDxY
iOS app PR 2994 build with same issue but with logs! https://app.testfairy.com/projects/4803590-status/builds/7470587/sessions/8/?accessToken=HuTz2mxKrTCnOEo2jcM2aFDATcQ
- Open Status
- Other 2 devices should add you are a contact and each of them sends "hello" to you 
- Open 2 new 1-1 chats, Add them as Contact, reply "hi" in each of the 2 chats
- kill the all
- launch app again
- Sign in
- Wait for 5 minutes on Chats screen, if app does not crash - open each of chats, return to Chats, switch to each tab Wallet, Discover, Contacts several times, kill app and log in again

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: 
release build 31 for 0.9.13 (not same as develop) 
Nightly builds from Jan 8,9 and 10
* Operating System: Android 6.0.1 real device Galaxy S6, iPhone 7 plus, iOS 11.0.3

**Comment by dmitryn** [comment]
Happens to me in silmulator as well, here what i see in `/data/thumbstones`:

```
01-10 13:07:35.188  4242  4286 D ReactNativeJS: DEBUG [status-im.protocol.web3.delivery:168] - :delivery-callback
01-10 13:07:35.190  4242  4286 D ReactNativeJS: DEBUG [status-im.utils.handlers:29] - Handling re-frame event:  :incoming-message
01-10 13:07:35.218  4242  4286 D ReactNativeJS: DEBUG [status-im.utils.handlers:29] - Handling re-frame event:  :update-message-status
01-10 13:07:36.156  4242  4242 W LocationServices: Location Permissions Not Granted Yet.  Try again after requesting.
01-10 13:07:36.156  4242  4242 I MapboxEventManager: Permissions not granted yet... let's try again in 30 seconds
01-10 13:07:36.163  4242  4595 D MapboxEventManager: No events in the queue to send so returning.
01-10 13:08:06.158  4242  4242 W LocationServices: Location Permissions Not Granted Yet.  Try again after requesting.
01-10 13:08:06.158  4242  4242 I MapboxEventManager: Permissions not granted yet... let's try again in 30 seconds
01-10 13:08:36.160  4242  4242 W LocationServices: Location Permissions Not Granted Yet.  Try again after requesting.
01-10 13:08:36.160  4242  4242 I MapboxEventManager: Permissions not granted yet... let's try again in 30 seconds
01-10 13:09:06.164  4242  4242 W LocationServices: Location Permissions Not Granted Yet.  Try again after requesting.
01-10 13:09:06.164  4242  4242 I MapboxEventManager: Permissions not granted yet... let's try again in 30 seconds
01-10 13:09:35.743  4242  4286 W google-breakpad: Microdump crash handler failed.
01-10 13:09:35.747  4242  4286 E libc    : Access denied finding property "ro.logd.timestamp"
01-10 13:09:35.747  4242  4286 E libc    : Access denied finding property "persist.logd.timestamp"
```

**Comment by annadanchenko** [comment]
On @oskarth request made extra investigation
1. Set Location = On in Android Os settings
2. On first run, after adding contacts, opened location command to get a promt and Allowed sharing location (but haven't sent any location to the chat)
3. Relaunched app waited about 2-5 minutes on Chats screen doing nothing with the phone -> app crashed (or went on background, so not visible on foreground and if switch to it then Sign in screen shown instead of Chats, so seems like Android OS kills it)
android: https://app.testfairy.com/projects/4803622-status/builds/7469768/sessions/14/?accessToken=qAQQ3F8r2C-u-N-JBUcvud7znI4
iOS (Still running ok) https://app.testfairy.com/projects/4803590-status/builds/7469771/sessions/5/?accessToken=kfIpJGZp2qIjnKj8GcNAPxBph/g
4. Performed same steps on same 3 devices using 0.9.12 release build, can run app for at least 10 minutes and it does not crash

**Comment by oskarth** [comment]
```
01-11 00:25:33.336  1702  1789 D ReactNativeJS: DEBUG [status-im.protocol.web3.delivery:167] - :delivery-callback
01-11 00:25:33.343  1702  1789 D ReactNativeJS: DEBUG [status-im.utils.handlers:29] - Handling re-frame event:  :incoming-message
01-11 00:25:57.599  1702  1789 D ReactNativeJS: DEBUG [status-im.protocol.web3.transport:15] - :post-message {:sig "0x0410792a0b73d411af3f7f76c5d36e093049085d4b1b8ed2311f286f5752244b66334eb467838e677cb6a484b1785eef5f6a86d7d66d50d4d9f893f4718bf2d91a", :topic "0xaabb11ee", :payload "0x7b3a6d6573736167652d69642022313531353634383130363436392d61383966326437612d313461322d353261642d383131312d336562376238653539306366222c203a72657175697265732d61636b3f20747275652c203a74797065203a6d6573736167652c203a74696d657374616d7020313531353634383130363436392c203a636f6e74656e742022686921222c203a636f6e74656e742d747970652022746578742f706c61696e222c203a636c6f636b2d76616c756520362c203a73686f773f20747275657d0a", :pubKey "0x04ea9469dd0a76322e02070b06cbb30d334b69e9e584712c445bbf68225868ce02b8b12835c4b43ab0fb0abf16e27bd8a13bbe7f0d23e32209679cfc8829becc19", :ttl 120}
01-11 00:25:57.600  1702  1789 D ReactNativeJS: DEBUG [status-im.protocol.web3.delivery:153] - :attempt-was-made 1515648106469-a89f2d7a-14a2-52ad-8111-3eb7b8e590cf
01-11 00:25:57.617  1702  1789 D ReactNativeJS: DEBUG [status-im.protocol.web3.delivery:167] - :delivery-callback
01-11 00:25:57.619  1702  1789 D ReactNativeJS: DEBUG [status-im.utils.handlers:29] - Handling re-frame event:  :incoming-message
01-11 00:25:57.747  1702  1789 D ReactNativeJS: DEBUG [status-im.protocol.discoveries:30] - :send-online
01-11 00:25:57.748  1702  1789 W ReactNativeJS: Setting a timer for a long period of time, i.e. multiple minutes, is a performance and correctness issue on Android as it keeps the timer module awake, and timers can only be called when the app is in the foreground. See https://github.com/facebook/react-native/issues/12981 for more info.
01-11 00:25:57.748  1702  1789 W ReactNativeJS: (Saw setTimeout with duration 180000ms)
01-11 00:25:57.750  1702  1789 D ReactNativeJS: DEBUG [status-im.protocol.web3.delivery:64] - :add-pending-message! {:from "0x0410792a0b73d411af3f7f76c5d36e093049085d4b1b8ed2311f286f5752244b66334eb467838e677cb6a484b1785eef5f6a86d7d66d50d4d9f893f4718bf2d91a", :message-id "1515648357746-3a8c24c8-5686-503a-86eb-88f65b24edbc", :keypair {:public "3840daf9cda072580c1ec6d82d507f1e3f8a0498ea1c48ed42b88705c2b1447939b799d79b607e365f833890adbe526aba54093ce7fd422eecca44ce9f63d0053b4724c22ff574c0f9824eebc66101cbc4eff4c2db308ed216a7c5d74b743be7afe", :private "38490d42d842679df8cd5daa79caff0860a97fbf881037b91d2143477c0e80b4946108c7dc54b2dbcaa3d1f2483190aae35"}, :requires-ack? false, :type :online, :key-password "status-discovery", :payload {:content {:timestamp 1515648357747}}, :topics ["0xaabb11ee"]}
```

EDIT: See https://github.com/status-im/status-react/issues/2408 and https://github.com/status-im/status-react/commit/e273568544ee507c65e582bbfeb918663ade746f

EDIT2: Consider verifying with https://github.com/status-im/status-react/pull/2853

**Comment by oskarth** [comment]
> iOS app PR 2994 build with same issue but with logs! https://app.testfairy.com/projects/4803590-status/builds/7470587/sessions/8/?accessToken=HuTz2mxKrTCnOEo2jcM2aFDATcQ

Two observations:
- logs stop after 00:25 for some reason
- (private) memory performance keeps increasing from 300mb (presumably) to 1900 at end of 6m, suggesting perhaps a memory leak

On the other hand, memory perf is barely changing (increasing a bit) on Android, so I don't know.

**Comment by oskarth** [comment]
Using nightly build 174 and three devices with fresh accounts: iPhone 6s, iPhone X and Galaxy S8.

- All have each other as contacts
- All send one message to all others
- Wait 5m on chat list screen (phone goes to background)
- No crash on any device

However:

- Go look at each chat and flip tabs, then kill app
- Wait 5m on chat list screen
- Only Galaxy S8 unresponsive for 20-30s then crashes, iPhones fine

UPDATE:
- If i just do flip tabs/kill app -> wait 5m on Galaxy S8 I can't reproduce twice in a row now.

**Comment by oskarth** [comment]
@rasom If you could have a look at this too it'd be great. I'm out of time and ideas right now. @yenda might try bisect.

**Comment by annadanchenko** [comment]
Logs from debugging by @rasom 
```
W/google-breakpad( 4391): ### ### ### ### ### ### ### ### ### ### ### ### ###
F/libc    ( 4391): Fatal signal 6 (SIGABRT), code -6 in tid 4591 (RenderThread)
I/DEBUG   (  316): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
I/DEBUG   (  316): Build fingerprint: 'samsung/trlteuc/trlteatt:5.0.1/LRX22C/N910AUCU1COC4:user/release-keys'
I/DEBUG   (  316): Revision: '12'
I/DEBUG   (  316): ABI: 'arm'
I/DEBUG   (  316): pid: 4391, tid: 4591, name: RenderThread  >>> im.status.ethereum <<<
I/DEBUG   (  316): signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
I/DEBUG   (  316): Abort message: 'Failed to make current on surface 0x988a7260, error=EGL_BAD_SURFACE'
I/DEBUG   (  316):     r0 00000000  r1 000011ef  r2 00000006  r3 00000000
I/DEBUG   (  316):     r4 afb7adb8  r5 00000006  r6 0000000c  r7 0000010c
I/DEBUG   (  316):     r8 b4e4ee2c  r9 afb7ad08  sl afb7ad00  fp b4e4ee68
I/DEBUG   (  316):     ip 000011ef  sp afb7a840  lr b6ea0ff5  pc b6ec4998  cpsr 600f0010
I/DEBUG   (  316):
I/DEBUG   (  316): backtrace:
I/DEBUG   (  316):     #00 pc 00037998  /system/lib/libc.so (tgkill+12)
I/DEBUG   (  316):     #01 pc 00013ff1  /system/lib/libc.so (pthread_kill+52)
I/DEBUG   (  316):     #02 pc 00014c0f  /system/lib/libc.so (raise+10)
I/DEBUG   (  316):     #03 pc 00011531  /system/lib/libc.so (__libc_android_abort+36)
I/DEBUG   (  316):     #04 pc 0000fcbc  /system/lib/libc.so (abort+4)
I/DEBUG   (  316):     #05 pc 00007709  /system/lib/libcutils.so (__android_log_assert+88)
I/DEBUG   (  316):     #06 pc 00043c17  /system/lib/libhwui.so
I/DEBUG   (  316):     #07 pc 000427a7  /system/lib/libhwui.so
I/DEBUG   (  316):     #08 pc 00042861  /system/lib/libhwui.so
I/DEBUG   (  316):     #09 pc 00044111  /system/lib/libhwui.so
I/DEBUG   (  316):     #10 pc 000440b9  /system/lib/libhwui.so
I/DEBUG   (  316):     #11 pc 00044a71  /system/lib/libhwui.so
I/DEBUG   (  316):     #12 pc 00045407  /system/lib/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+66)
I/DEBUG   (  316):     #13 pc 0000ef11  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+112)
I/DEBUG   (  316):     #14 pc 00060401  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+72)
I/DEBUG   (  316):     #15 pc 0000ea81  /system/lib/libutils.so
I/DEBUG   (  316):     #16 pc 000137bb  /system/lib/libc.so (__pthread_start(void*)+30)
I/DEBUG   (  316):     #17 pc 0001189b  /system/lib/libc.so (__start_thread+6) 
```


```
E/libutils.threads( 9897): (android threadPriority=0)
F/libc    ( 9897): Fatal signal 7 (SIGBUS), code -6, fault addr 0x26a9 in tid 10590 (Chrome_InProcRe)
I/DEBUG   (  316): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
I/DEBUG   (  316): Build fingerprint: 'samsung/trlteuc/trlteatt:5.0.1/LRX22C/N910AUCU1COC4:user/release-keys'
I/DEBUG   (  316): Revision: '12'
I/DEBUG   (  316): ABI: 'arm'
I/DEBUG   (  316): pid: 9897, tid: 10590, name: Chrome_InProcRe  >>> im.status.ethereum <<<
I/DEBUG   (  316): signal 7 (SIGBUS), code -6 (SI_TKILL), fault addr 0xa3cfdbc2
I/DEBUG   (  316):     r0 0000005c  r1 2b600000  r2 00000000  r3 000007bf
I/DEBUG   (  316):     r4 a4763346  r5 09538000  r6 00000000  r7 06a2f148
I/DEBUG   (  316):     r8 06a2fc00  r9 06a2f9f8  sl 00000001  fp 09538000
I/DEBUG   (  316):     ip a4ca1ce0  sp 06a2f120  lr a33e81f1  pc a3cfdbc2  cpsr 000b0030
I/DEBUG   (  316):
I/DEBUG   (  316): backtrace:
I/DEBUG   (  316):     #00 pc 01733bc2  /data/app/com.google.android.webview-1/lib/arm/libwebviewchromium.so
F/google-breakpad(10603): Microdump skipped (uninteresting)
```

```
I/VideoCapabilities( 9897): Unsupported profile 4 for video/mp4v-es
I/v8      ( 9897):
I/v8      ( 9897): <--- Last few GCs --->
I/v8      ( 9897):
I/v8      ( 9897):
I/v8      ( 9897): <--- JS stacktrace --->
I/v8      ( 9897):
W/libc    ( 9897): pthread_create failed: couldn't allocate 1064960-byte stack: Out of memory
E/art     ( 9897): Throwing OutOfMemoryError "pthread_create (1040KB stack) failed: Try again"
```

**Comment by annadanchenko** [comment]
from crashing session android: [log_jan11.txt](https://github.com/status-im/status-react/files/1623800/log_jan11.txt)

**Comment by mandrigin** [comment]
I managed to run the app under Instruments, but, frankly no real breakthrough there.
Some of my findings (iOS Simulator, iOS 11.2):
1. STR. For me, it was enough to receive 1-2 messages while being on the "Chats" tab. I joined a public channel (`#status`), restarted the app and it crashed.
2. It is a memory leak. I managed to grow the app to over 3GB in size on my iOS simulator.
3. It looks like the chat list is unable to render the incoming messages when they arrive and freezes there.
4. It is almost certainly a React-side problem. Together with the memory growth, CPU was running hot by 5 JS threads (WebKit's working threads).
5. For some reasons, most of the buttons are unclickable when the problem reproduces. But I can still switch tabs by _swiping_ between them.

Otherwise, I tried to use Chrome Web Dev Tools, but nothing interesting there... I'll try to dig a bit more tomorrow if it won't be solved.
